core/src/wallet/ripple/RippleLikeAccount.cpp

```diff
@@ -0,0 +1,414 @@
+                                                        }
+                                                    });
+
+            auto startTime = DateUtils::now();
+            eventPublisher->postSticky(
+                    std::make_shared<Event>(api::EventCode::SYNCHRONIZATION_STARTED, api::DynamicObject::newInstance()),
+                    0);
+            future.onComplete(getContext(), [eventPublisher, self, startTime](const Try<Unit> &result) {
+                api::EventCode code;
+                auto payload = std::make_shared<DynamicObject>();
+                auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(
+                        DateUtils::now() - startTime).count();
+                payload->putLong(api::Account::EV_SYNC_DURATION_MS, duration);
+                if (result.isSuccess()) {
+                    code = api::EventCode::SYNCHRONIZATION_SUCCEED;
+                } else {
+                    code = api::EventCode::SYNCHRONIZATION_FAILED;
+                    payload->putString(api::Account::EV_SYNC_ERROR_CODE,
+                                       api::to_string(result.getFailure().getErrorCode()));
+                    payload->putInt(api::Account::EV_SYNC_ERROR_CODE_INT, (int32_t) result.getFailure().getErrorCode());
+                    payload->putString(api::Account::EV_SYNC_ERROR_MESSAGE, result.getFailure().getMessage());
+                }
+                eventPublisher->postSticky(std::make_shared<Event>(code, payload), 0);
+                std::lock_guard<std::mutex> lock(self->_synchronizationLock);
+                self->_currentSyncEventBus = nullptr;
+
+            });
+            return eventPublisher->getEventBus();
+        }
+
+        std::shared_ptr<RippleLikeAccount> RippleLikeAccount::getSelf() {
+            return std::dynamic_pointer_cast<RippleLikeAccount>(shared_from_this());
+        }
+
+        void RippleLikeAccount::startBlockchainObservation() {
+            _observer->registerAccount(getSelf());
+        }
+
+        void RippleLikeAccount::stopBlockchainObservation() {
+            _observer->unregisterAccount(getSelf());
+        }
+
+        bool RippleLikeAccount::isObservingBlockchain() {
+            return _observer->isRegistered(getSelf());
+        }
+
+        std::string RippleLikeAccount::getRestoreKey() {
+            return _keychain->getRestoreKey();
+        }
+
+        void RippleLikeAccount::broadcastRawTransaction(const std::vector<uint8_t> &transaction,
+                                                        const std::shared_ptr<api::StringCallback> &callback) {
+            _explorer->pushTransaction(transaction).map<std::string>(getContext(),
+                                                                     [](const String &seq) -> std::string {
+                                                                         //TODO: optimistic update
```

Comment:
I suggest to remove the TODO by transforming it into a Jira issue. :) (also, I have no idea what optimistic update is)

Reply:
I created JIRA task, but I leave the comment (don't forget that we are open source :trollface: )